static/js/zamboni/files.js

```diff
@@ -46,161 +196,72 @@ if (typeof SyntaxHighlighter !== 'undefined') {
-        if (this.getParam('light') == true)
-            this.params.toolbar = this.params.gutter = false;
-
-        if (this.getParam('collapse') == true)
-            classes.push('collapsed');
-
-        classes.push('nogutter');
-
-        // add custom user style name
-        classes.push(this.getParam('class-name'));
-
-        // add brush alias to the class name for custom CSS
-        classes.push(this.getParam('brush'));
-
-        lineNumbers = this.figureOutLineNumbers(code);
 
         // find matches in the code using brushes regex list
-        matches = this.findMatches(this.regexList, code);
-        // processes found matches into the html
-        html = this.getMatchesHtml(code, matches);
-        // finally, split all lines so that they wrap well
-        html = this.getCodeLinesHtml(html, lineNumbers);
-
-        // finally, process the links
-        if (this.getParam('auto-links'))
-            html = processUrls(html);
+        var matches = this.findMatches(this.regexList, code);
 
-        if (typeof(navigator) != 'undefined' && navigator.userAgent && navigator.userAgent.match(/MSIE/))
-            classes.push('ie');
+        // processes found matches into the html
+        var html = this.getMatchesHtml(code, matches);
 
-        var line_order = Math.ceil(Math.log(lineNumbers.length) / Math.log(10)) + 1;
-        var lines_width = 1.2 * line_order + 4;  // 1.2 ex width per digit, just to be safe, + 4 for padding.
+        // N.B.(Kris): This... unbelievably... trims whitespace from both
+        // sides of the output, even though it was already trimmed
+        // prior to output in the stock version of this function.
+        html = '|' + html;
+        this.getCodeLinesHtml(html);
 
-        if (SyntaxHighlighter.amo_vars.lines) {
-            // Just generating HTML for the diff highlighter.
-            // Bail early.
-            return '';
-        }
+        // And we're done. The line HTML is now in `SyntaxHighlighter.amo_vars.lines`.
+    };
 
-        html =
-            '<div id="highlighter_' + this.id + '" class="' + classes.join(' ') + '">'
-                + (this.getParam('toolbar') ? sh.toolbar.getHtml(this) : '')
-                    + '<table border="0" cellpadding="0" cellspacing="0">'
-                        + this.getTitleHtml(this.getParam('title'))
-                            + '<colgroup><col class="highlighter-column-line-numbers" style="width: ' + lines_width + 'ex;"/>'
-                                      + '<col class="highlighter-column-code"/></colgroup>'
-                            + '<tbody>'
-                                + html
-                            + '</tbody>'
-                    + '</table>'
-            + '</div>';
-
-        return html;
+    // Urgh. Why, oh why, with the absurd Crockford Closures...
+    // I shouldn't have to do this.
+    //
+    // Slightly modified from the built-in version to handle
+    // newer keywords:
+    new function() {
```

Comment:
It's basically a hack to get a scope for variables. There are three common ways of doing this, these days: `(function() { ... })();` `!function() { ... }()`, and `new function() { ... }`. I tend to prefer the latter, since it puts all of the information up front.